From 8cba1d2a8182f19b0ce664da020426c26f142a4f Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Wed, 3 May 2017 08:29:55 +0200
Subject: [PATCH 19/84] fstools: snapshot: handle jffs2 conversion using
 upgraded

We can reuse the kill_remaining and run_ramfs facilities of the stage2 run
by upgraded.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>
---
 package/system/fstools/Makefile       |  2 +-
 package/system/fstools/files/snapshot | 14 ++++++++------
 2 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/package/system/fstools/Makefile b/package/system/fstools/Makefile
index 6f1f511d38..f2e0580912 100644
--- a/package/system/fstools/Makefile
+++ b/package/system/fstools/Makefile
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=fstools
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(LEDE_GIT)/project/fstools.git
diff --git a/package/system/fstools/files/snapshot b/package/system/fstools/files/snapshot
index baf24f1e3e..a495e34345 100644
--- a/package/system/fstools/files/snapshot
+++ b/package/system/fstools/files/snapshot
@@ -64,14 +64,16 @@ do_convert_jffs2() {
 do_convert() {
 	. /lib/functions.sh
 	. /lib/upgrade/common.sh
-	ubus call system upgrade
-	touch /tmp/sysupgrade
+
 	cd /overlay/upper
 	tar czf /tmp/snapshot.tar.gz *
-	kill_remaining TERM
-	sleep 3
-	kill_remaining KILL
-	run_ramfs '. /sbin/snapshot; do_convert_jffs2'
+
+	install_bin /sbin/upgraded
+	ubus call system sysupgrade "{
+		\"prefix\": \"$RAM_ROOT\",
+		\"path\": \"\",
+		\"command\": \". /sbin/snapshot; do_convert_jffs2\"
+	}"
 }
 
 [ -n "$(cat /proc/mounts|grep /overlay|grep jffs2)" ] && {
-- 
2.17.1

